cli/cmd/plugin/ui/web/tanzu-ui/src/shared/utilities/Array.util.ts

```diff
@@ -12,6 +12,22 @@ export function middle<ARRAY_TYPE>(source: ARRAY_TYPE[]): ARRAY_TYPE | undefined
     return elementAt(source, index);
 }
 
+/**
+ * Find the first element in list2 which is also in list1.
+ * @param list1 is a list of string.
+ * @param list2 contains 0 or 1 element which is in list1.
+ * @returns string is the element in both list1 and list2.
+ */
+export function findFirstOverlappingElement(list1: string[], list2: string[]): string {
+    const set = new Set(list1);
+    for (let i = 0; i < list2.length; i++) {
+        if (set.has(list2[i])) {
+            return list2[i];
+        }
+    }
+    return list1[Math.round(list1.length / 2)];
```

Comment:
Should move this piece of strategy out of this method, and implement near the caller

Should warn to console if match is not found

Reply:
moved